lib/rubygems/specification.rb

```diff
@@ -1752,10 +1752,11 @@ def dependencies
   #
   #   [depending_gem, dependency, [list_of_gems_that_satisfy_dependency]]
 
-  def dependent_gems
+  def dependent_gems(check_dev=true)
```

Comment:
Shouldn't this default to false if the problem is that we are always checking for development even without the flag?

Reply:
I kept the method behaving the same as before although a quick check shows that it's only used by the uninstaller call I'm changing, so I guess we could change the default value or even make it mandatory.

I just figured this was the safest thing to do since it doesn't change previous calls to the method without arguments.

Reply:
Note that the problem is not how the command behaves (that is correct), it's the information it gives that it's incorrect.

For example, in the example I gave, with the current version of the code, if I removed the 5 runtime dependants of `rake`, it would uninstall properly without a prompt even if it's telling me that I have dozens of conflicts.